evm_arithmetization/src/witness/transition.rs

```diff
@@ -297,12 +299,30 @@ pub(crate) fn log_kernel_instruction<F: Field>(state: &mut GenerationState<F>, o
     assert!(pc < KERNEL.code.len(), "Kernel PC is out of range: {}", pc);
 }
 
-pub(crate) trait Transition<F: Field>: State<F> {
+pub(crate) trait Transition<F: Field>: State<F>
+where
+    Self: Sized,
+{
     /// When in jumpdest analysis, adds the offset `dst` to the jumpdest table.
     /// Returns a boolean indicating whether we are running the jumpdest
     /// analysis.
     fn generate_jumpdest_analysis(&mut self, dst: usize) -> bool;
 
+    fn final_exception(&mut self) -> anyhow::Result<()> {
+        let exc_code: u8 = 6;
```

Comment:
`6` is duplicated in different places, we may want to use a global CONST to prevent any possible issues